### PR TITLE
[TOOLS-3116] Open task card by task link before taskboard

### PR DESCRIPTION
**Pull request by AndreKrebs**

### CODE REVIEW

File: core/src/main/resources/static/scripts/taskboard.js [hunk 1]
```diff
@@ -4,6 +4,12 @@ function Taskboard() {
     var _cardFieldFilters;
     var _issuesBySteps;
     var _laneConfiguration;
+
```
Comment (objective-solutions-user) [inline core/src/main/resources/static/scripts/taskboard.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3ATrailingWhitespace)

File: core/src/main/resources/static/scripts/taskboard.js [hunk 2]
```diff
@@ -4,6 +4,12 @@ function Taskboard() {
     var _cardFieldFilters;
     var _issuesBySteps;
     var _laneConfiguration;
+    
+    var _tShirtSizesResolver;
+
```
Comment (objective-solutions-user) [inline core/src/main/resources/static/scripts/taskboard.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3ATrailingWhitespace)

File: core/src/main/resources/static/scripts/taskboard.js [hunk 3]
```diff
@@ -145,6 +151,9 @@ function Taskboard() {
     };
 
     this.findIssueByKey = function(issueKey){
+    	if (!this.issues)
```
Comment (objective-solutions-user) [inline core/src/main/resources/static/scripts/taskboard.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3ATrailingWhitespace)

File: core/src/main/resources/static/scripts/taskboard.js [hunk 4]
```diff
@@ -145,6 +151,9 @@ function Taskboard() {
     };
 
     this.findIssueByKey = function(issueKey){
+    	if (!this.issues) 
+    		return undefined;
+
```
Comment (objective-solutions-user) [inline core/src/main/resources/static/scripts/taskboard.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3ATrailingWhitespace)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 4 issues
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: cardboard/src/main/resources/static/elements/taskboard/taskboard-home.html [hunk 5]
```diff
@@ -580,7 +580,7 @@
                             taskboard.setLaneConfiguration(data.laneConfiguration);
                             taskboard.urlJira = data.urlJira;
                             taskboard.urlLinkGraph = data.urlLinkGraph;
-                            taskboard.setTshirtSizes(data.tShirtSizes);
+                            taskboard.promiseSetTshirtSizes(data.tShirtSizes);
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/taskboard-home.html]: Que tal "resolveTshirtSizesPromise"?

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail.html [hunk 6]
```diff
@@ -1365,7 +1365,13 @@
 
                 attached: function() {
                     var self = this;
-                    this.tShirtSizes = taskboard.tShirtSizes;
+
+                    taskboard.getTshirtSizes().then(function(value){
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/issue-detail.html]: Mudar de "value" para tshirtSizes.

File: cardboard/src/main/resources/static/elements/taskboard/subtask-item-form.html [hunk 7]
```diff
@@ -126,7 +126,9 @@
             },
 
             attached: function() {
-                this.tShirtSizeOptions = taskboard.getTshirtSizes();
+                taskboard.getTshirtSizes().then(function(value){
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/subtask-item-form.html]: Mudar de "value" para tshirtSizes.

File: core/src/main/resources/static/scripts/taskboard.js [hunk 8]
```diff
@@ -5,6 +5,12 @@ function Taskboard() {
     var _issuesBySteps;
     var _laneConfiguration;
 
+    var _tShirtSizesResolver;
+
+    var _tShirtSizes = new Promise(function(resolve) {
+    	_tShirtSizesResolver = resolve;
```
Comment (jhonypg) [inline core/src/main/resources/static/scripts/taskboard.js]: Ajusta a indentação para 4 spaces. Acho que o mesmo ocorre nas linhas de mudanças abaixo, que estão usando tab.

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail.html (hunk 6 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/issue-detail.html]: Feito!

File: cardboard/src/main/resources/static/elements/taskboard/subtask-item-form.html (hunk 7 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/subtask-item-form.html]: Feito!

File: core/src/main/resources/static/scripts/taskboard.js (hunk 8 above)
Comment (AndreKrebs) [reply core/src/main/resources/static/scripts/taskboard.js]: Feito!

File: cardboard/src/main/resources/static/elements/taskboard/taskboard-home.html (hunk 5 above)
Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/taskboard-home.html]: ok!

**Pull request merged**